book_source/06_reference/04_docker/03-quickstart.Rmd

```diff
@@ -0,0 +1,78 @@
+# Quickstart for Docker and PEcAn.
+
+This is a short documentation on how to start with Docker and PEcAn. This will not go into much detail about about how to use docker.
+
+## Install Docker
+
+You will need to install docker first. See https://www.docker.com/community-edition#/download
+
+##Running PEcAn
+
+We will use a yaml file to describe the containers that we make up, This yaml file is called [docker-compose.yml](https://github.com/PecanProject/pecan/blob/master/docker-compose.yml). Before you get started you will need to download this file first. Place this file somewhere on your machine and cd to the folder that contains this file.
+
+Next we will start the docker containers to start Docker. Since both BETY and PEcAn depend on a populated database we will first start the database and populate it with some data. This only needs to be done once, if this is done already, you can skip this step and go to the next step
+
+```bash
+docker-compose -p pecan up -d postgres
+docker run -ti --rm \
+	--network pecan_pecan \
+	hub.ncsa.illinois.edu/pecan/bety:latest initialize
+docker run -ti --rm \
+	--network pecan_pecan \
+    --volume pecan_pecan:/data \
+    hub.ncsa.illinois.edu/pecan/data:develop
+```
+
+If this is finished (or you had done this in the past), we can start the full stack:
+
+```bash
+docker-compose -p pecan up -d
+```
+
+This will start all the containers marked green in the architecture diagram. Once this is done you have a working instance of PEcAn. Following is a list of URL's you can visit
+
+| URL                          | Service                                               |
+| ---------------------------- | ----------------------------------------------------- |
+| http://localhost:8000/pecan/ | PEcAn web GUI                                         |
+| http://localhost:8000/bety/  | BETY web application                                  |
+| http://localhost:8000/minio/ | File browser (username carya, password illinois)      |
+| http://localhost:8000/       | RabbitMQ (username guest, password guest)             |
+| http://localhost:8001/       | Træfik, webserver shows mapping from url to container |
+
+## Testing PEcAn
+
+To test PEcAn you can use the following curl statement, or use the webpage to submit a request:
+
+```bash
+curl -v -X POST \
+    -F 'hostname=docker' \
+    -F 'modelid=5000000002' \
+    -F 'sitegroupid=1' \
+    -F 'siteid=772' \
+    -F 'sitename=Niwot Ridge Forest/LTER NWT1 (US-NR1)' \
+    -F 'pft[]=temperate.coniferous' \
+    -F 'start=2004/01/01' \
+    -F 'end=2004/12/31' \
+    -F 'input_met=5000000005' \
+    -F 'email=' \
+    -F 'notes=' \
+    'http://localhost:8000/pecan/04-runpecan.php'
```

Comment:
```
$ curl -v -X POST \ 
        -F 'hostname=docker' \
        -F 'modelid=5000000002' \
        -F 'sitegroupid=1' \
        -F 'siteid=772' \
        -F 'sitename=Niwot Ridge Forest/LTER NWT1 (US-NR1)' \
        -F 'pft[]=temperate.coniferous' \
        -F 'start=2004/01/01' \
        -F 'end=2004/12/31' \
        -F 'input_met=5000000005' \
        -F 'email=' \
        -F 'notes=' \
        'http://localhost:8000/pecan/04-runpecan.php'

Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8000 (#0)
> POST /pecan/04-runpecan.php HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.54.0
> Accept: */*
> Content-Length: 1204
> Expect: 100-continue
> Content-Type: multipart/form-data; boundary=------------------------67f9075f5ab8d6a1
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 307 Temporary Redirect
< Cache-Control: no-store, no-cache, must-revalidate
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< Date: Sun, 17 Jun 2018 19:26:37 GMT
< Expires: Thu, 19 Nov 1981 08:52:00 GMT
< Location: http://localhost:8000/setups/edit.php?key=database&message=1
< Pragma: no-cache
< Server: Apache/2.4.25 (Debian)
< Set-Cookie: PHPSESSID=40b34535c8908d7d48749e2255036bd5; path=/
< X-Powered-By: PHP/7.2.6
* HTTP error before end of send, stop sending
< 
* Closing connection 0
```

Reply:
Update: Error above was when using Homebrew docker-machine with xhyve driver. Redid the quickstart steps with official Docker For Mac and now can't reproduce, so not sure if Homebrew issue or if I missed a step the first time.

Reply:
http://localhost:8000/setups/edit.php?key=database&message=1 this means that it could not connect to the database. So it looks like there is some issue with the networking in that case.